src/components/common/Settings.tsx

```diff
@@ -51,81 +54,81 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
-const Settings = () => {
+const Settings: FC = () => {
   const [open, setOpen] = useState(false);
-  const classes = useStyles();
+  const { classes } = useStyles();
   const { t } = useTranslation();
-  const { data: settings, isLoading } = useAppSettings();
-  const { mutate: postAppSetting } = useMutation(
-    MUTATION_KEYS.POST_APP_SETTING,
-  );
-  const { mutate: patchAppSetting } = useMutation(
-    MUTATION_KEYS.PATCH_APP_SETTING,
-  );
+  const {
+    appSettingArray: settings,
+    postAppSetting,
+    patchAppSetting,
+  } = useAppSettingContext();
 
-  const saveSettings = (originalSetting, newSetting) => {
-    if (originalSetting?.id) {
-      patchAppSetting(newSetting);
-    } else {
-      postAppSetting(newSetting);
+  interface NewSetting {
+    id?: AppSetting['id'];
+    name?: AppSetting['name'];
+    data: AppSetting['data'];
+  }
+  const saveSettings = (newSetting: NewSetting): void => {
```

Comment:
```suggestion
  const saveSettings = (newSetting: Pick<AppSetting, 'id' | 'name' | 'data'>): void => {
```

Reply:
Not sure it would work. `id` and `name` are optional.

Reply:
Pick is just a way of creating a new type from a subset of properties of another type, this means you would get:
```ts
type NewSetting = {
  id?: string;
  name?: string;
  data: whatever;
```